aristanetworks/ansible-avd
# Feat: Add AVD Example single-dc-l3ls

## Change Summary



Ansible AVD example for a single DC using L3LS and eAPI.

## Proposed changes

This example contains what is needed to run Ansible AVD against a network topology for a single DC using L3LS and eAPI.

Topology:

- 2 spines.
- 2 sets of layer 3 leaf switches.
- 2 layer 2 leaf switches.

Included in the file structure:
- `basic-switch-configurations` folder containing basic switch configs to be applied to switches before to running Ansible AVD ensuring basic connectivity.
- `group_vars` folder containing all relevant group variable files including definition of fabric-wide services (VRFs and VLANs) as well as connectivity to two dual-attached servers (one per DC).
- `intended` folder configurations (both structured and actual EOS CLI configs).
- `documentation` folder with device-specific and fabric-wide documentation.
- `config_backup` folder with EOS CLI config backups.
- `inventory.yml` describing the Ansible inventory.
- `playbook.yml` describing the Ansible playbook to generate structured and actual EOS CLI configs and push configuration to switches using eAPI.
- `ansible.cfg` describing basic Ansible settings.
- `README.md` describing everything about the example, how to use it etc.

## How to test


1) Set up a topology identical to the one described using identical hostnames, interface connections, IP addresses etc.
The example can be tested either using physical or virtual topologies.
Please note that the example is aimed at being tested using a virtual topology, specifically using vEOS-lab. The two servers can also be simulated using vEOS-lab, but the relevant configuration ***isn't*** included in this example.
If a physical topology is used, please note the vEOS-lab specific settings documented inline in the YAML files as well as in the README.MD file included. These must be removed or changed to ensure optimum functionality in a physical topology.

2) Disable pushing to the switches in the `playbook.yml` file by commenting out the last role as shown below:

```yml
---
- name: Run AVD
  hosts: FABRIC
  gather_facts: false
  tasks:
    - name: generate intended variables
      import_role:
        name: arista.avd.eos_designs
    - name: generate device intended config and documentation
      import_role:
        name: arista.avd.eos_cli_config_gen
    # - name: deploy configuration to device
    #   import_role:
    #      name: arista.avd.eos_config_deploy_eapi
```

When running the playbook you will only update the local generated content (configs, documentation etc.), which makes it possible to play with this example without any virtual or physical network topology.



Virtual topology created in GNS3 with vEOS-lab 4.26.5M on all devices.
Basic connectivity tests from server to server using the different VRFs and VLANs.

## Checklist

### Repository Checklist
